### PR TITLE
Payless

### LINKED ISSUE #526: Payless Shoes
https://www.payless.com/on/demandware.store/Sites-payless-Site/default/Stores-GetNearestStores?postalCode=54701&countryCode=US&distanceUnit=imperial&maxdistance=30

### PR DESCRIPTION
**Pull request by evanmosseri**
Fixes #526

### CODE REVIEW

File: locations/spiders/toysrus.py [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import scrapy
-from scrapy.spiders import BaseSpider
```
Comment (iandees) [inline locations/spiders/toysrus.py]: Please limit your changes in a pull request to a single spider.

File: locations/spiders/payless.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def start_requests(self):
+        urls = (
+            'https://www.payless.com/on/demandware.store/Sites-payless-Site/default/Stores-GetNearestStores?postalCode'
+            '=11230&countryCode=US&distanceUnit=imperial&maxdistance=5000',
+        )
+        for url in urls:
+            yield scrapy.Request(url=url, callback=self.parse, headers=default_headers)
+
+    def parse(self, response):
+        if response.status == 200:
```
Comment (iandees) [inline locations/spiders/payless.py]: There's no need to do explicit checks on HTTP response code like this. Just let the spider fail if the `json.loads()` call fails.

File: locations/spiders/payless.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            stores = json.loads(response.body_as_unicode())
+            for store in stores["stores"].values():
+                street = "{} {}".format(store["address1"], store["address2"]).strip()
+                has_house_number = store["address1"].split(" ")[0].isnumeric()
+                website = "https://www.payless.com/on/demandware.store/Sites-payless-Site/default/Stores-Details" \
+                          "?StoreID={}".format(
+                    store["number"])
+                yield GeojsonPointItem(lat=store["latitude"],
+                                       lon=store["longitude"],
```
Comment (iandees) [inline locations/spiders/payless.py]: Can you clean this up a bit so the arguments for `GeojsonPointItem` aren't indented so much? Maybe knock it down to the next line or build up a `props` dictionary and pass it in.

File: locations/spiders/payless.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import scrapy
+from locations.items import GeojsonPointItem
+import json
+import re
+import json
+
+default_headers = {
+    "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/537.36 (KHTML, like Gecko) "
```
Comment (iandees) [inline locations/spiders/payless.py]: Does the spider work without a fake user agent? I'd prefer to use a descriptive user agent if possible.

Comment (evanmosseri) [reply locations/spiders/payless.py]: Sorry, force of habit :) What kind of user agent would be appropriate? Something like "alltheplacesbot"?

File: locations/spiders/payless.py (hunk 2 above)
Comment (evanmosseri) [reply locations/spiders/payless.py]: kk

File: locations/spiders/payless.py (hunk 3 above)
Comment (evanmosseri) [reply locations/spiders/payless.py]: Sure

File: locations/spiders/payless.py (hunk 4 above)
Comment (iandees) [reply locations/spiders/payless.py]: Understand. The default is [set here](https://github.com/alltheplaces/alltheplaces/blob/master/locations/settings.py#L20) so you don't need to set anything in your spider.

**Comment by iandees** [comment]
Please don't dismiss/delete my review comments.

**Comment by evanmosseri** [comment]
Sorry, what comments are you referring to? If you expand the reviews that have "Hide outdated" in them, they are all there.

**Comment by iandees** [comment]
Hum, I think I'm misinterpreting [this message](https://github.com/alltheplaces/alltheplaces/pull/560#event-1397172864) from GitHub. I thought it indicated you were deleting my comments but I think it means you're clicking some other button. Nevermind!

**Pull request merged**